### DISCUSSION TITLE
Equip 3rd party types with a deconstructor

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#101] [General]
# Equip 3rd party types with a deconstructor

Deconstructor are a C# language feature that allows you to define a method that will be called when an object is being split up into its components. While this is straightforward to implement for your own types, it is not possible to add a deconstructor to a 3rd party type - or is it?

https://steven-giesel.com/blogPost/0775d3d3-8f90-4546-95d8-71a6b1e7b0e8/equip-3rd-party-types-with-a-deconstructor

**Comment by toupswork** [comment]
Yeah I love this. It would be nice if it worked like in JavaScript where you don't have to specify every variable. Sure you can use a discard character, but that's just messy for long props.
```
const { prop } = { prop:0, prop1:1, prop2:2 }
```

**Reply by linkdotnet** [reply]
I do love this notation as well - and it comes out of the box thanks to the structured typing.

you could add a bunch of overloads in c# to somewhat mimic that behavior but it is by far not as powerful.

**Comment by toupswork** [comment]
Do you need ideas for future blogs? Here's one: `ConditionalWeakTable`

**Reply by linkdotnet** [reply]
I do have a backlog and this sounds like a good idea! I did work with it once some time back ago.
